spec/inputs/rabbitmq_spec.rb

```diff
@@ -166,21 +166,73 @@ def spawn_and_wait(instance)
   end
 
   describe "receiving a message with a queue specified" do
-    let(:queue_name) { "foo_queue" }
+    let(:queue_name) { "logstash-input-rabbitmq-#{rand(0xFFFFFFFF)}" }
     let(:config) { super.merge("queue" => queue_name) }
+    let(:queue) { test_channel.queue(queue_name, :auto_delete => true) }
 
     it "should process the message" do
       message = "Foo Message"
-      q = test_channel.queue(queue_name)
-      q.publish(message)
+      queue.publish(message)
 
       event = output_queue.pop
       expect(event["message"]).to eql(message)
     end
+
+    it "should save message properties into a @metadata field" do
+      epoch = 1454789198
```

Comment:
no special way needed. My understanding is that RSpec seeds random at startup and all you do is just call rand() or similar like you would normally.